tests/unit/test_job.py

```diff
@@ -4812,6 +4811,58 @@ def test_result_with_max_results(self):
             tabledata_list_request[1]["query_params"]["maxResults"], max_results
         )
 
+    def test_result_w_retry(self):
+        from google.cloud.bigquery.table import RowIterator
+
+        query_resource = {
+            "jobComplete": False,
+            "jobReference": {"projectId": self.PROJECT, "jobId": self.JOB_ID},
+        }
+        query_resource_done = {
+            "jobComplete": True,
+            "jobReference": {"projectId": self.PROJECT, "jobId": self.JOB_ID},
+            "schema": {"fields": [{"name": "col1", "type": "STRING"}]},
+            "totalRows": "2",
+        }
+        job_resource = self._make_resource(started=True)
+        job_resource_done = self._make_resource(started=True, ended=True)
+        job_resource_done["configuration"]["query"]["destinationTable"] = {
+            "projectId": "dest-project",
+            "datasetId": "dest_dataset",
+            "tableId": "dest_table",
+        }
+
+        connection = _make_connection(
+            exceptions.NotFound("not normally retriable"),
+            query_resource,
+            query_resource_done,
```

Comment:
We had a few places where custom retry was already supported. I'd like to see a few more exceptions to be totally sure that all API calls are getting the right retry object.

```suggestion
            exceptions.NotFound("not normally retriable"),
            query_resource_done,
            exceptions.NotFound("not normally retriable"),
```